Jenkinsfile

```diff
@@ -137,23 +137,34 @@ def checkout_git(label) {
         cd cryptography
         git fetch origin +refs/pull/${env.CHANGE_ID}/merge:
         git checkout -qf FETCH_HEAD
-        git rev-parse HEAD
         """
+        if (label.contains("windows")) {
+            bat script
+        } else {
+            sh """#!/bin/sh
+                set -xe
+                ${script}
+            """
+        }
     } else {
-        script = """
-        git clone --depth=1 https://github.com/pyca/cryptography.git cryptography
-        cd cryptography
-        git checkout ${env.BRANCH_NAME}
-        git rev-parse HEAD
-        """
-    }
-    if (label.contains("windows")) {
-        bat script
-    } else {
-        sh """#!/bin/sh
-        set -xe
-        ${script}
-        """
+        checkout([
+            $class: 'GitSCM',
+            branches: [[name: "*/${env.BRANCH_NAME}"]],
+            doGenerateSubmoduleConfigurations: false,
+            extensions: [[
+                $class: 'RelativeTargetDirectory',
+                relativeTargetDir: 'cryptography'
+            ]],
+            submoduleCfg: [],
+            userRemoteConfigs: [[
+                'url': 'https://github.com/pyca/cryptography'
+            ]]
+        ])
+        if (label.contains("windows")) {
+            bat "git rev-parse HEAD"
+        } else {
+            sh "git rev-parse HEAD"
+        }
```

Comment:
This needs to be pulled out of this branch. Also you'll need to `cd cryptography` before you `rev-parse`, I assume?